controller/telemetry/server.go

```diff
@@ -249,6 +218,27 @@ func (s *server) ListPods(ctx context.Context, req *read.ListPodsRequest) (*publ
 		return nil, err
 	}
 
+	// Reports is a map from instance name to the absolute time of the most recent
+	// report from that instance.
+	reports := make(map[string]time.Time)
+	// Query Prometheus for reports in the last 30 seconds.
+	res, err := s.prometheusAPI.Query(ctx, requestsMetric + "[30s]", time.Time{})
+	if err != nil {
+		return nil, err
+	}
+	if res.Type() != model.ValMatrix {
+		err = fmt.Errorf("Unexpected query result type (expected Matrix): %s", res.Type())
+		log.Errorf("%s", err)
+		return nil, err
+	}
+	for _, s := range res.(model.Matrix) {
+		labels := metricToMap(s.Metric)
+		timestamp := s.Values[len(s.Values)-1].Timestamp
+		secs := timestamp / 1000
+		nanos := (timestamp % 1000) * 1000000
+		reports[labels["pod"]] = time.Unix(int64(secs), int64(nanos))
```

Comment:
There's no need to split the timestamp when calling `time.Unix` -- from the godoc:

> It is valid to pass nsec outside the range [0, 999999999]

So you could instead write this as:

```
		reports[labels["pod"]] = time.Unix(0, int64(timestamp)*int64(time.Millisecond))
```

I also recommend using the [duration constants](https://golang.org/pkg/time/#Duration), instead of hardcoding 1000, 1000000, etc.